### PR TITLE
Introduce User entity

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
Adding `user.Info` entity

**Why?**
`user.Info` will be used to authenticate user and generate JWT token for authorization 

**How did you test it?**

**Potential risks**
None

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/user/info.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// SOFTWARE.
+
+package user
+
+// Info holds information about user
+// This information is used to generate corresponding JWT claim
+type Info struct {
+	Name     string
+	Password string
+	Admin    bool
```
Comment (davidporter-id-au) [inline common/user/info.go:30]: Same as before, role, probably a string or enum is fine. Enum requires making IDL changes every time there's a role change, which may be a annoying, but is clearer. String allows for finer and faster iteration on what roles may actually do.

File: common/user/info.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Groups   []string
```
Comment (davidporter-id-au) [inline common/user/info.go:31]: What is group?

**Review by davidporter-id-au — APPROVED** [review]

**Review by davidporter-id-au — CHANGES_REQUESTED** [review]
sorry, I meant to comment before but accidentally hit approve, I do think the role/group thing needs clarification.

File: common/user/info.go (hunk 2 above)
Comment (mantas-sidlauskas) [reply common/user/info.go:31]: In Cadence we have OAuth authorization already implemented. This struct mainly reflects model for authorization already in use.

On server side, you can update domain metadata with keys: READ_GROUPS and WRITE_GROUPS to define what "group" has what permissions. Then groups and admin flag are checked in JWT claim verification part when "access controlled handler" is used.

References to get more context: 
* [JWT Claim struct](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/common/authorization/oauthAuthorizer.go#L48)
* [JWT claim verification](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/common/authorization/oauthAuthorizer.go#L148)
* [Access controlled handler authorize flow example](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/service/frontend/accessControlledHandler.go#L747)

**Pull request closed without merging**